### PR TITLE
[ONNX] Extend ONNX Frontend with Function `Celu-12`

### LINKED ISSUE #20543: [Good First Issue]: Extend ONNX Frontend with Function Celu-12
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires extending OpenVINO ONNX Frontend with Function Celu.
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Celu)
More details can be found in [ONNX Changelog](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Celu-12)
1. Create `.hpp` and `.cpp` files for *Windows [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op)													
2. Prepare an implementation of this operator in form of a function. It should be placed in [opset 1 namespace](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/unique.hpp#L13).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/15076
- https://github.com/openvinotoolkit/openvino/pull/14936
- https://github.com/openvinotoolkit/openvino/pull/12510
- https://github.com/openvinotoolkit/openvino/pull/19428

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by siddhant-0707**
### Details:
 - created implementation for `Celu-12` operator
 - created tests

### Tickets:
 - Closes #20543

### CODE REVIEW

**Comment by siddhant-0707** [comment]
@gkrivor

**Comment by gkrivor** [comment]
@siddhant-0707 you should remove it from the list, otherwise it tries to expand it using an existing simple operators, if possible.

**Comment by andrei-kochin** [comment]
build_jenkins

**Comment by gkrivor** [comment]
build_jenkins

**Comment by siddhant-0707** [comment]
It appears `Celu` only takes in float type input, so the `int` test I created previously is irrelevant. Also added check to confirm type.

File: src/frontends/onnx/frontend/src/op/celu.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+namespace onnx_import {
+namespace op {
+namespace set_1 {
+OutputVector celu(const Node& node) {
+    double alpha = node.get_attribute_value<float>("alpha", 1.0);
+    auto alpha_node = default_opset::Constant::create(element::f32, Shape{}, {alpha});
+    auto zero_node = default_opset::Constant::create(element::f32, Shape{}, {0});
+    auto one_node = default_opset::Constant::create(element::f32, Shape{}, {1});
+    auto input = node.get_ng_inputs().at(0);
+    CHECK_VALID_NODE(node, input.get_element_type() == element::f32, "Only float32 input type is supported")
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/op/celu.cpp:23]: Please remove this check, supported types are validated in the constructors of ov operators, also such restriction can be not backward compatible.

**Review by mitruska — CHANGES_REQUESTED** [review]
Proposed subgraph looks mathematically correct, but is bigger and not aligned with ONNX Function Celu decomposition.

As currently ONNX Celu op is expanded via ONNX standard mechanism to `Div->Elu->Mul`, 
changes in the subgraph produced by OV can affect performance and backward compatibility. 

I suggest to align the subgraph with the ONNX standard decomposition.

**Review by mitruska — COMMENTED** [review]
Thank you @siddhant-0707 for applying the requested changes. 
Just few additional suggestions for the updated code.

File: src/frontends/onnx/frontend/src/op/celu.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+namespace onnx {
+namespace op {
+namespace set_1 {
+ov::OutputVector celu(const ov::frontend::onnx::Node& node) {
+    double alpha = node.get_attribute_value<float>("alpha", 1.0);
+    auto alpha_node = v0::Constant::create(element::f32, ov::Shape{}, {alpha});
+    auto x_celu = node.get_ov_inputs().at(0);
+    x_celu = std::make_shared<v0::Convert>(x_celu, element::f32);
+
+    auto divide_node = std::make_shared<v1::Divide>(x_celu, alpha_node);
```
[the remark this answers is not in the document]
Comment (siddhant-0707) [reply src/frontends/onnx/frontend/src/op/celu.cpp]: Hey, @mitruska the type alignment logic you suggested needs to be in a separate PR right? Removing the Convert statement for now. BTW, thank you so much for the detailed review!

**Comment by mitruska** [comment]
build_jenkins

**Review by gkrivor — APPROVED** [review]

**Comment by mitruska** [comment]
build_jenkins

**Comment by gkrivor** [comment]
build_jenkins

**Pull request merged**